packages/react-native-editor/package.json

```diff
@@ -108,6 +108,8 @@
 		"test": "cross-env NODE_ENV=test jest --verbose --config ../../test/native/jest.config.js",
 		"test:debug": "cross-env NODE_ENV=test node --inspect-brk ../../node_modules/.bin/jest --runInBand --verbose --config ../../test/native/jest.config.js",
 		"test:update": "npm run test -- --updateSnapshot",
+		"test:perf": "cross-env NODE_ENV=test TEST_RUNNER_PATH=../../node_modules/.bin/jest TEST_RUNNER_ARGS='--runInBand --testMatch \"**/*.perf-native.[jt]s?(x)\" --verbose --config ../../test/native/jest.config.js' reassure",
```

Comment:
What is the rationale behind the `perf-native.js` suffix? How much work would it be to use a different folder name for this type of tests like `benchmark` or something similar?

Reply:
`'**/test/*.[jt]s?(x)',` - is my biggest concern as it might also recognize the file name as a unit test for web code.

> I'm not too sure why this project's preset configuration supports both __tests__ (Jest default) and test (Gutenberg customization).

I don't remember exactly now, but I belive this pattern existed before Jest got integrated with Gutenberg.

> is your proposal to match **/benchmark/*.test.js for native performance tests?

You will also find `benchmark` folders, too. `performance` work for me.

Reply:
> `'**/test/*.[jt]s?(x)',` - is my biggest concern as it might also recognize the file name as a unit test for web code.

@gziolo ah, I see. The following configuration excludes any native-specific test files, at least in the context of the `test:unit` script. 

https://github.com/WordPress/gutenberg/blob/77feb6093dbdcc46e8d77d25299a877191deacf7/test/unit/jest.config.js#L35

Although, my guess is that the current exclusion of the _new performance tests_ is accidental and originates from the lack of escaping the period character.

```diff
diff --git a/test/unit/jest.config.js b/test/unit/jest.config.js
index 9d19a5c9fe..39764020cb 100644
--- a/test/unit/jest.config.js
+++ b/test/unit/jest.config.js
@@ -32,7 +32,7 @@ module.exports = {
 		'<rootDir>/.*/build-module/',
 		'<rootDir>/.*/build-types/',
 		'<rootDir>/.+.d.ts$',
-		'<rootDir>/.+.native.js$',
+		'<rootDir>/.+\\.native.js$',
 		'/packages/react-native-*',
 	],
 	resolver: '<rootDir>/test/unit/scripts/resolver.js',
```

Regardless, updating our approach to utilize a `performance` directory works for me. An example mobile performance test file would be `packages/block-library/src/paragraph/performance/paragraph.native.js`. 

Note: the file would _not include_ a `test` suffix.

Reply:
We also use pattern matching for linting, for example:

https://github.com/WordPress/gutenberg/blob/trunk/.eslintrc.js#L369L383

It isn't impossible to make it work with exceptions like currently but it looks like we could avoid it with the `/performance/` subdirectory.

Reply:
@gziolo @dcalhoun Thanks for the feedback. Based on the discussion, I've moved the `*.perf-native.js` performance tests to their own colocated `/performance/` subdirectory, so the updated directory path would be `packages/block-library/src/paragraph/performance/paragraph.perf-native.js`.

Based on [this above comment](https://github.com/WordPress/gutenberg/pull/49221/files/3903b6ff635c5693386a0fe8f5698e4aa39d14e0#r1147869546), however, is the consensus that we want to use the extension `*.native.js` in place of `*.perf-test.js` as well? I am fine with also making this change, but just seeking to clarify if it's still necessary with the added `/performance/` directory.

Reply:
Great, makes sense. I've updated the file extensions to `*.native.js`.